game_calls.cpp

```diff
@@ -1,14 +1,25 @@
-#include "game_calls.h"
+#include <algorithm>
 #include <iostream>
 #include <limits>
 #include <ctime>
+#include "game_calls.h"
+
+Code::Code()
+{
+	// Easier to write! Arguments: number of elements, value of each element
+	code = std::vector<int>(4, 0);
+}
+
+Code::~Code() = default;		// .clear() will be called by ~vector()
 
 void Code::UpdateGuess()
 {
-	srand(time(NULL));
+	// Try using std::random instead
+	//srand(time(NULL));		// Compiler throws a warning here; so explicitly cast it
+	srand(static_cast<unsigned int>(time(NULL)));
 	for (size_t i = 0; i < code.size(); ++i)
 	{
-		code[i] = rand() % 10;
+		code[i] = rand() % 10;	// This is a flawed approach BTW
```

Comment:
How do you mean?

Reply:
The distribution is non uniform. Say `rand()` returns 0-255, then 256-260 will never be encountered, skewing the `% 10` result towards 0-5.